File: extract-css.py
import os
path = r'C:\Users\SHUBHAM\Desktop\inFraVex'
with open(os.path.join(path, 'index.html'), 'r', encoding='utf-8') as f:
    html = f.read()
start = html.find('<style>')
end = html.find('</style>')
if start == -1 or end == -1:
    print('No <style> tag found')
    exit()
css = html[start+7:end].strip()
with open(os.path.join(path, 'main.css'), 'w', encoding='utf-8') as f:
    f.write(css)
new_html = html[:start] + '<link rel="stylesheet" href="main.css" />' + html[end+8:]
with open(os.path.join(path, 'index.html'), 'w', encoding='utf-8') as f:
    f.write(new_html)
print(f'Done. CSS: {len(css)} chars. HTML: {len(new_html)} chars')

File: extract-css.js
const fs = require('fs');
const html = fs.readFileSync('index.html', 'utf8');
const styleStart = html.indexOf('<style>');
const styleEnd = html.indexOf('</style>');
if (styleStart === -1 || styleEnd === -1) {
  console.log('No <style> tag found - may already be extracted');
  process.exit(0);
}
const css = html.substring(styleStart + 7, styleEnd).trim();
fs.writeFileSync('main.css', css, 'utf8');
const newHtml = html.substring(0, styleStart) + '<link rel="stylesheet" href="main.css" />' + html.substring(styleEnd + 8);
fs.writeFileSync('index.html', newHtml, 'utf8');
console.log('Done. CSS:', css.length, 'chars extracted. HTML:', newHtml.length, 'chars remaining.');
